### PR TITLE
[#3818][#4100] Improvement(client-python): Replace all assert in client-python & fix: Validate methods of request in client-python are not correct

### PR DESCRIPTION
**Pull request by jingjia88**
### What changes were proposed in this pull request?

- Fix `validate` errors (attributes are not allowed to be None) in `catalog_update_request.py` .
- Unify the implementation way of `validate` : using raise. 
- Replace all `assert` with defined exception.

### Why are the changes needed?

Fix: #3818, #4100 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

test with` ./gradlew client:client-python:test`

### CODE REVIEW

**Comment by jerryshao** [comment]
@noidname01 would you please help to review this?

**Comment by jerryshao** [comment]
@jingjia88 would you please fix the conflicts and continue the work as @noidname01 mentioned, thanks.

**Comment by jerryshao** [comment]
We're planning to kick off the 0.6.0 release this week, hopes this can be done this week @jingjia88 , thanks a lot.

**Comment by jingjia88** [comment]
@noidname01   I’ve addressed issue #4100 in this PR as well. Could you please review the modifications? Thank you!

**Comment by noidname01** [comment]
> @noidname01 I’ve addressed issue #4100 in this PR as well. Could you please review the modifications? Thank you!

@jingjia88 Sure, I'll review it ASAP.

File: clients/client-python/gravitino/catalog/base_schema_catalog.py [hunk 1]
```diff
@@ -238,12 +239,13 @@ def validate(self):
...
-            self.name() is not None and len(self.name().strip()) > 0
-        ), "name must not be blank"
-        assert self.type() is not None, "type must not be None"
-        assert (
-            self.provider() is not None and len(self.provider().strip()) > 0
-        ), "provider must not be blank"
-        assert self.audit_info() is not None, "audit must not be None"
+        if self.rest_client is None:
+            raise IllegalArgumentException("restClient must be set")
+        if not (self.name() and len(self.name().strip()) > 0):
```
Comment (noidname01) [inline clients/client-python/gravitino/catalog/base_schema_catalog.py:244]: I think here can modify into `if not self.name() or not self.name.strip()`

File: clients/client-python/gravitino/catalog/base_schema_catalog.py [hunk 2]
```diff
@@ -238,12 +239,13 @@ def validate(self):
...
+            raise IllegalArgumentException("name must not be blank")
+        if self.type() is None:
+            raise IllegalArgumentException("type must not be None")
+        if not (self.provider() and len(self.provider().strip()) > 0):
```
Comment (noidname01) [inline clients/client-python/gravitino/catalog/base_schema_catalog.py:248]: Same here

**Review by noidname01 — CHANGES_REQUESTED** [review]
Except for minor issues, LGTM

File: clients/client-python/gravitino/catalog/base_schema_catalog.py (hunk 2 above)
Comment (jingjia88) [reply clients/client-python/gravitino/catalog/base_schema_catalog.py]: Done!

File: clients/client-python/gravitino/catalog/base_schema_catalog.py (hunk 1 above)
Comment (jingjia88) [reply clients/client-python/gravitino/catalog/base_schema_catalog.py]: Done!

**Comment by jerryshao** [comment]
@noidname01 can you please take another review when you have time, thanks!

Generally the PR looks good to me.

**Review by noidname01 — APPROVED** [review]

**Review by jerryshao — APPROVED** [review]

**Pull request merged**